mysql-test/suite/innodb/t/innodb-index.test

```diff
@@ -1202,3 +1202,13 @@ call mtr.add_suppression("Plugin 'InnoDB' init function returned error");
 call mtr.add_suppression("Plugin 'InnoDB' registration as a STORAGE ENGINE failed");
 
 --enable_query_log
+
+--echo #
+--echo # MDEV-22855 Assertion (!field->prefix_len ||
+--echo #	field->fixed_len == field->prefix_len)
+--echo #	failed in btr_node_ptr_max_size
+--echo #
+CREATE TABLE t1(c CHAR(255) CHARACTER SET UTF32, KEY k1(c(250)))ENGINE=InnoDB;
+INSERT INTO t1 VALUES(100000);
```

Comment:
This test is a little obfuscated: we are implicitly converting the numeric constant into a string literal `'100000'`. I would suggest just writing the following, if it reproduces the problem:
```
INSERT INTO t1 SET c='';
```
Also, can you try to replace the 255 and 250 with the minimum needed sizes to trigger the bug?